src/TimeoutMigrationTool/RabbitMq/RabbitMqWriter.cs

```diff
@@ -1,16 +1,132 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using RabbitMQ.Client;
 
 namespace Particular.TimeoutMigrationTool.RabbitMq
 {
     class RabbitMqWriter
     {
         public Task<bool> WriteTimeoutsTo(string rabbitMqBroker, List<TimeoutData> timeouts, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            using (var connection = GetConnection(rabbitMqBroker))
+            {
+                using (var model = connection.CreateModel())
+                {
+                    foreach (var timeout in timeouts)
+                    {
+                        PublishTimeout(model, timeout);
+                    }
+                }
+            }
         }
+
+        private void PublishTimeout(IModel model, TimeoutData timeout)
+        {
+            int level;
+
+            //TODO: guard for negative timespan
+            var delay = (timeout.Time - DateTime.UtcNow); 
+            var delayInSeconds = Convert.ToInt32(Math.Ceiling(delay.TotalSeconds));
+            var routingKey = CalculateRoutingKey(delayInSeconds, timeout.Destination, out level);
+            
+            var properties = model.CreateBasicProperties();
+            Fill(properties, timeout, delay);
+
+            //TODO: Blow up when the time > 8.5 years[max value].
+
+            model.BasicPublish(LevelName(level), routingKey, true, properties, timeout.State);
+        }
+
+        private IConnection GetConnection(string rabbitMqBroker)
+        {
+            var factory = new ConnectionFactory();
+            factory.Uri = new Uri(rabbitMqBroker);
+            return factory.CreateConnection();
+        }
+
+        public static void Fill(this IBasicProperties properties, TimeoutData timeout, TimeSpan delay)
+        {
+            properties.MessageId = timeout.Id; //TODO: pull from the header first 
```

Comment:
This is what core does when receiving a message https://github.com/Particular/NServiceBus/blob/300e511294d6694f22abc8b7340cce6753ad434d/src/NServiceBus.Core/Transports/IncomingMessage.cs#L23 so not sure if we need to replicate that here or not?

Reply:
Yeah I will replicate it, just to be on the safe side